src/Acuminator/Acuminator.Tests/Tests/Utilities/AttributeInformation/AttributeInformationTests.cs

```diff
@@ -61,51 +63,101 @@ private async Task TestAttributeInformationAsync(string source, List<bool> expec
-		public void TestAreBoundAggregateRecursiveAttribute(string source) =>
-			_testIsBoundAttribute(source, new List<bool> { false, true });
+		public async Task TestAreBoundAggregateRecursiveAttributeAsync(string source) =>
+			await _testIsBoundAttributeAsync(source, new List<BoundFlag> { BoundFlag.Unbound, BoundFlag.DbBound });
 
-		private void _testIsBoundAttribute(string source, List<bool> expected)
+		private async Task _testIsBoundAttributeAsync(string source, List<BoundFlag> expected)
 		{
 			Document document = CreateDocument(source);
-			SemanticModel semanticModel = document.GetSemanticModelAsync().Result;
+			SemanticModel semanticModel = await document.GetSemanticModelAsync();
 			var syntaxRoot = document.GetSyntaxRootAsync().Result;
 
-			List<bool> actual = new List<bool>();
+			List<BoundFlag> actual = new List<BoundFlag>();
 			var pxContext = new PXContext(semanticModel.Compilation);
 
 			var properties = syntaxRoot.DescendantNodes().OfType<PropertyDeclarationSyntax>();
 
+			var attributeInformation = new Acuminator.Utilities.Roslyn.PXFieldAttributes.AttributeInformation(pxContext);
+
 			foreach (var property in properties)
 			{
 				var typeSymbol = semanticModel.GetDeclaredSymbol(property);
 				var attributes = typeSymbol.GetAttributes();
 				foreach (var attribute in attributes)
 				{
-					var attributeInformation = new Acuminator.Utilities.Roslyn.PXFieldAttributes.AttributeInformation(pxContext);
-					actual.Add(attributeInformation.IsBoundAttribute(attribute.AttributeClass));
+					actual.Add(attributeInformation.IsBoundAttribute(attribute));
 				}
 			}
 			Assert.Equal(expected, actual);
 		}
 
+		[Theory]
+		[EmbeddedFileData(@"PropertyIsDBBoundFieldAttribute.cs")]
+		public async Task FieldBoundAttributesWithDynamicIsDBFieldSetInConstructorAsync(string source) =>
+			await _testIsDBFieldPropertyAsync(source,
+								   new List<BoundFlag> { BoundFlag.DbBound, BoundFlag.Unbound });
+
+		[Theory]
+		[EmbeddedFileData(@"PropertyIsDBBoundFieldWithDefinedAttributes.cs")]
+		public async Task FieldBoundAttributesWithDynamicIsDBFieldSetInAttributeDefinitionAsync(string source) =>
+		   await _testIsDBFieldPropertyAsync(source,
+								  new List<BoundFlag> { BoundFlag.Unknown, BoundFlag.Unknown, BoundFlag.Unknown, BoundFlag.Unknown });
+
+		[Theory]
+		[EmbeddedFileData(@"PropertyIsDBBoundFieldWithoutDefinedAttributes.cs", internalCodeFileNames: new string[] { @"ExternalAttributes1.cs", @"ExternalAttributes2.cs" })]
+		public async Task FieldBoundAttributesWithDynamicIsDBFieldSetInExternalAttributeDefinitionAsync(string source, string externalAttribute1, string externalAttribute2) =>
+		   await _testIsDBFieldPropertyAsync(source,
+								  new List<BoundFlag> { BoundFlag.Unknown, BoundFlag.Unknown },
+								  new string[] { externalAttribute1, externalAttribute2 });
+
+		private async Task _testIsDBFieldPropertyAsync(string source, List<BoundFlag> expected, string[] code = null)
```

Comment:
Private methods must be named in PascalCase.
See default C# naming conventions.